orc8r/cloud/go/services/orchestrator/servicers/protected/indexer_servicer.go

```diff
@@ -117,39 +118,48 @@ func setSecondaryStates(ctx context.Context, networkID string, states state_type
 	if len(sessionIDToIMSI) == 0 && len(cTeidToHwId) == 0 {
 		return stateErrors, nil
 	}
-	multiError := merrors.NewMulti()
+	errs := &multierror.Error{}
 	if len(sessionIDToIMSI) != 0 {
 		err := directoryd.MapSessionIDsToIMSIs(ctx, networkID, sessionIDToIMSI)
-		multiError = multiError.AddFmt(err, "failed to update directoryd mapping of session IDs to IMSIs %+v", sessionIDToIMSI)
+		if err != nil {
```

Comment:
i know this has been mentioned below, but just as a placeholder: remove all ifs since `Append` will ignore nils

Reply:
this contains formatting, the `errors. Errorf` result will still not be `nil`, even if `err` was `nil`.

But this 3 line block could be extracted to a function, since they just differ in a few words.

Reply:
Introduced a wrapper in this file.